### PR TITLE
Adds circleci 2.0

### LINKED ISSUE #264: CircleCI 2.0
Update to use CircleCI 2.0

### PR DESCRIPTION
**Pull request by christinach**
Fixes #264

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    steps:
+      - checkout
+      - run: mkdir -p $CIRCLE_ARTIFACTS $CIRCLE_TEST_REPORTS
+      # Dependencies
+      # Restore the dependency cache
+      - type: cache-restore
+        name: Restore bundle cache
+        key: cicognara-rails-{{ checksum "Gemfile.lock" }}
+
+      # - run: sudo update-alternatives --set java /usr/lib/jvm/jdk1.8.0/bin/java
```
Comment (tpendragon) [inline .circleci/config.yml]: Can you get rid of these commented lines?

Comment (christinach) [reply .circleci/config.yml]: yes, I was still working on that. sorry.

Comment (christinach) [reply .circleci/config.yml]: Do you want the to run the rake test_server separately and include the jetty_wait script?

Comment (tpendragon) [reply .circleci/config.yml]: It'd be a little faster, but I think this is fine the way it is.

Comment (christinach) [reply .circleci/config.yml]: ok

**Pull request merged**